### PR TITLE
support for writable external tables with PXF

### PR DESCRIPTION
**Pull request by denalex**

### CODE REVIEW

File: concourse/scripts/regression_tests_pxf.bash [hunk 1]
```diff
@@ -17,11 +17,30 @@ function run_regression_test() {
...
+	fi
+
+	# verify data written by PXF
+	cat /tmp/pxf-write/* | sort
+	num_rows=\$(cat /tmp/pxf-write/* | wc -l)
+	if [ "\$num_rows" != "100" ]; then
+	    echo "ERROR: Unexpected number of rows written by PXF = \$num_rows"
+	    exit 1
+	fi
+
```
Comment (lavjain) [inline concourse/scripts/regression_tests_pxf.bash]: This shell script should just run the test. Data verification should be done as part of pg_regress.

File: gpAux/extensions/pxf/src/pxfbridge.c [hunk 2]
```diff
@@ -121,6 +157,38 @@ build_uri_for_read(gphadoop_context *context)
...
 
 /*
+ * Format the URI for writing by adding PXF service endpoint details
+ */
+static void
+build_uri_for_write(gphadoop_context *context)
+{
+	appendStringInfo(&context->uri, "http://%s:%d/%s/%s/Writable/stream?path=%s",
+					 PxfDefaultHost, PxfDefaultPort,
+					 PXF_SERVICE_PREFIX, PXF_VERSION, context->write_file_name.data);
```
Comment (lavjain) [inline gpAux/extensions/pxf/src/pxfbridge.c]: Please check if we can use data->authority (to be consistent with read api).

File: gpAux/extensions/pxf/src/pxfprotocol.c [hunk 3]
```diff
@@ -65,22 +65,16 @@ pxfprotocol_validate_urls(PG_FUNCTION_ARGS)
 	elog(DEBUG2, "pxfprotocol_validate_urls: uri %s", uri_string);
 	GPHDUri    *uri = parseGPHDUri(uri_string);
 
-	/* Test that Fragmenter or Profile was specified in the URI */
-	if (!GPHDUri_opt_exists(uri, FRAGMENTER) && !GPHDUri_opt_exists(uri, PXF_PROFILE))
```
Comment (shivzone) [inline gpAux/extensions/pxf/src/pxfprotocol.c]: We still need this validation for external readable tables right ? Also, usage of Profiles will be the recommended way to read/write data using pxf protocol

File: concourse/scripts/regression_tests_pxf.bash (hunk 1 above)
Comment (denalex) [reply concourse/scripts/regression_tests_pxf.bash]: ok, thanks, will try to implement this in pg_regress

File: gpAux/extensions/pxf/src/pxfprotocol.c (hunk 3 above)
Comment (denalex) [reply gpAux/extensions/pxf/src/pxfprotocol.c]: yes, we need this logic. but it already exists below (lines 78-old or 72-new). The removed condition was a duplication with a bug for write cases.

**Pull request merged**